pkg/application-grid-controller/controller/service/util/service_util.go

```diff
@@ -60,16 +71,27 @@ func constructServicePortIdentify(port *corev1.ServicePort) string {
 }
 
 func KeepConsistence(sg *crdv1.ServiceGrid, svc *corev1.Service) *corev1.Service {
+	// NEVER modify objects from the store. It's a read-only, local cache.
+	// You can use DeepCopy() to make a deep copy of original object and modify this copy
+	// Or create a copy manually for better performance
 	copyObj := svc.DeepCopy()
-	if copyObj.Labels == nil {
-		copyObj.Labels = make(map[string]string)
+	// Append existed ServiceGrid labels to service to be checked
+	if sg.Labels != nil {
+		copyObj.Labels = sg.Labels
+		copyObj.Labels[common.GridSelectorName] = sg.Name
+		copyObj.Labels[common.GridSelectorUniqKeyName] = sg.Spec.GridUniqKey
+	} else {
+		copyObj.Labels = map[string]string{
+			common.GridSelectorName:        sg.Name,
+			common.GridSelectorUniqKeyName: sg.Spec.GridUniqKey,
```

Comment:
Now deployment, statefulset, service all have GridSelectorUniqKeyName to display uniqkey, maybe we can make it as a convention; While different with others ,service have TopologyAnnotationsKey in Annotations before, it have the same effect with GridSelectorUniqKeyName, we can remove it.

Reply:
I recommend to remove it when application-grid-wrapper develops more maturity or deprecated in future.